plugins/modules/azure_rm_virtualnetworkgatewayconnection.py

```diff
@@ -0,0 +1,515 @@
+        tags:
+            description:
+                - The resource tags.
+            type: dict
+            returned: always
+            sample: {'key1': 'value1'}
+        traffic_selector_policies:
+            description:
+                - The Traffic Selector Policies to be considered by this connection.
+            type: list
+            returned: always
+            sample: []
+        use_local_azure_ip_address:
+            description:
+                - Use private local Azure IP for the connection.
+            type: bool
+            returned: always
+            sample: false
+        use_policy_based_traffic_selectors:
+            description:
+                - Enable policy-based traffic selectors.
+            type: bool
+            returned: always
+            sample: false
+        virtual_network_gateway1:
+            description:
+                - The reference to virtual network gateway resource.
+            type: complex
+            returned: always
+            contains:
+                id:
+                    description:
+                        - The ID of the virtul network gateway resource.
+                    type: str
+                    returned: always
+                    sample: /subscriptions/xxx-xxx/resourceGroups/rg/providers/Microsoft.Network/virtualNetworkGateways/fredvng
+        virtual_network_gateway2:
+            description:
+                - The reference to virtual network gateway resource.
+            type: complex
+            returned: always
+            contains:
+                id:
+                    description:
+                        - The ID of the virtul network gateway resource.
+                    type: str
+                    returned: always
+                    sample: /subscriptions/xxx-xxx/resourceGroups/rg/providers/Microsoft.Network/virtualNetworkGateways/fredvng02
+
+'''
+
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
+try:
+    from azure.core.exceptions import HttpResponseError
+    from azure.core.polling import LROPoller
+except Exception:
+    # handled in azure_rm_common
+    pass
+
+
+class AzureRMVirutalNetworkGatewayConnection(AzureRMModuleBase):
+    """Utility class to get Azure Kubernetes Service Credentials facts"""
+
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            resource_group=dict(type='str', required=True),
+            name=dict(type='str', required=True),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            location=dict(type='str'),
+            virtual_network_gateway1=dict(type='str', required=False),
+            virtual_network_gateway2=dict(type='str'),
+            local_network_gateway2=dict(type='str'),
+            authorization_key=dict(type='str', no_log=True),
+            connection_type=dict(type='str', choices=["IPsec", "Vnet2Vnet", "ExpressRoute", "VPNClient"]),
```

Comment:
A local gateway is required when type is 'IPSec', while the second virtual gateway is required when type is 'Vnet2Vnet'; so I'd add here:

        required_if = [
            ('connection_type', 'IPsec', ['name', 'local_network_gateway2']),
            ('connection_type', 'Vnet2Vnet', ['name', 'virtual_network_gateway2'])
        ]
and then pass it to the super contructor

Reply:
@guidograzioli This is very good advice, I will update this PR, thank you very much!